modules/free/arrow-free-data/src/test/kotlin/arrow/free/FreeTest.kt

```diff
@@ -56,18 +64,21 @@ class FreeTest : UnitSpec() {
 
     val EQ: FreeEq<ForOps, ForId, Int> = Free.eq(IdMonad, idInterpreter)
 
+    @Suppress("UNCHECKED_CAST")
     testLaws(
       EqLaws.laws(EQ) { Ops.value(it) },
       MonadLaws.laws(Ops, EQ),
-      MonadLaws.laws(Free.monad(), EQ)
+      MonadLaws.laws(Free.monad(), EQ),
+      FoldableLaws.laws(Free.foldable(), { Ops.value(it) }, Eq.any()),
+      TraverseLaws.laws(Free.traverse(Id.functor()), Free.functor(), { a -> Ops.value(a) as Kind<FreePartialOf<ForId>, Int> }, Eq.any())
```

Comment:
@nomisRev I am having Type inference issues here is there a way to write down the laws without the UncheckedCast